docs/tutorials/layers_crf.ipynb

```diff
@@ -0,0 +1,1203 @@
+        "Create a list to convert tag ids to tag text."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-NChlWMnUTol"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "['<PAD>', 'O', 'B-PER', 'I-PER', 'B-ORG', 'I-ORG', 'B-LOC', 'I-LOC', 'B-MISC', 'I-MISC']\n"
+          ]
+        }
+      ],
+      "source": [
+        "tags = ['<PAD>'] + raw_tags\n",
+        "print(tags)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "d45nlEDXWqG6"
+      },
+      "source": [
+        "Define some constants which will be used in later."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "9A2UNwDnaPkX"
+      },
+      "outputs": [],
+      "source": [
+        "TAG_SIZE = len(tags)\n",
+        "VOCAB_SIZE = 20000"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "P8XXrsg1Wvvb"
+      },
+      "source": [
+        "Building vocabulary lookup layer for tokens."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "l3WJsqUBxpFX"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "20000\n",
+            "['[MASK]', '[UNK]', 'the', '.', ',', 'of', 'in', 'to', 'a', 'and']\n"
+          ]
+        }
+      ],
+      "source": [
+        "train_tokens = tf.ragged.constant(conll_data[\"train\"][\"tokens\"])\n",
+        "train_tokens = tf.map_fn(tf.strings.lower, train_tokens)\n",
+        "\n",
+        "lookup_layer = tf.keras.layers.StringLookup(max_tokens=VOCAB_SIZE, mask_token=\"[MASK]\", oov_token=\"[UNK]\")\n",
+        "lookup_layer.adapt(train_tokens)\n",
+        "\n",
+        "print(len(lookup_layer.get_vocabulary()))\n",
+        "print(lookup_layer.get_vocabulary()[:10])"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "lanszT8uXA6y"
+      },
+      "source": [
+        "Creating raw (without preprocess) train and validation dataset."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "j-u70kbtTSGg"
+      },
+      "outputs": [],
+      "source": [
+        "def create_data_generator(dataset):\n",
+        "  def data_generator():\n",
+        "    for item in dataset:\n",
+        "      yield item['tokens'], item['ner_tags']\n",
+        "  \n",
+        "  return data_generator\n",
+        "\n",
+        "data_signature= (\n",
+        "        tf.TensorSpec(shape=(None,), dtype=tf.string),\n",
+        "        tf.TensorSpec(shape=(None, ), dtype=tf.int32)\n",
+        ")\n",
+        "\n",
+        "train_data = tf.data.Dataset.from_generator(\n",
```

Comment:
I am not sure doing shuffle is a good idea. Since shuffle comes with randomness, it may make the model result unpredictable. On the other hand, it's a tutorial on CRF, and I am trying to make things simple. But if you consist that we need do a shuffle, that is OK. I can fix it in the following commit.